auction/contracts/AuctionHouse.sol

```diff
@@ -0,0 +1,338 @@
+            if (bidData.payouts.length == 0){
+                LibPart.Part[] memory payout = new LibPart.Part[](1);
+                payout[0].account = payable(buyer);
+                payout[0].value = 10000;
+                bidData.payouts = payout;
+            }
+            LibOrderDataV2.DataV2 memory aucData = LibAucDataV1.getPaymentData(currentAuction.data, currentAuction.dataType);
+            if (aucData.payouts.length == 0){
+                LibPart.Part[] memory payout = new LibPart.Part[](1);
+                payout[0].account = payable(seller);
+                payout[0].value = 10000;
+                aucData.payouts = payout;
+            }
+            doTransfersWithFees(
+                currentAuction.lastBid.amount, 
+                address(this), 
+                MatchFees(currentAuction.protocolFee, currentAuction.protocolFee, LibFeeSide.FeeSide.TAKE),
+                bidData, 
+                aucData, 
+                currentAuction.buyAsset, 
+                currentAuction.sellAsset.assetType, 
+                TO_MAKER
+            );
+            transferPayouts(currentAuction.sellAsset.assetType, currentAuction.sellAsset.value, address(this), bidData.payouts, TO_TAKER);
+        } else {
+            transfer(currentAuction.sellAsset, address(this), seller, TO_SELLER, UNLOCK); //nft back to seller
+        }
+    }
+
+    /// @dev returns true if auction started and hasn't finished yet, false otherwise
+    function checkAuctionRangeTime(uint _auctionId) public view returns (bool){
+        uint endTime = auctions[_auctionId].endTime;
+        uint currentTime = block.timestamp;
+
+        LibAucDataV1.DataV1 memory aucData = LibAucDataV1.parse(auctions[_auctionId].data, auctions[_auctionId].dataType);
+        if (aucData.startTime > 0 && aucData.startTime > currentTime) {
+            return false;
+        }
+        if (endTime > 0 && endTime < currentTime){
+            return false;
+        }
+        
+        return true;
+    }
+
+    /// @dev deletes auction after finalizing
+    function deactivateAuction(uint _auctionId) internal {
+        emit AuctionFinished(_auctionId, auctions[_auctionId]);
+        deleteAuctionForToken(auctions[_auctionId].sellAsset);
+        delete auctions[_auctionId];
+    }
+
+    /// @dev cancels existing auction without bid
+    function cancel(uint _auctionId) external {
+        require(checkAuctionExistence(_auctionId), "there is no auction with this id");
+        Auction storage currentAuction = auctions[_auctionId];
+        address seller = currentAuction.seller;
+        require(seller == _msgSender(), "auction owner not detected");
+        address buyer = currentAuction.buyer;
+        require(buyer == address(0), "can't cancel auction with bid");
+        transfer(currentAuction.sellAsset, address(this), seller, TO_SELLER, UNLOCK); //nft transfer back to seller
+        deactivateAuction(_auctionId);
+        AuctionCancelled(_auctionId);
```

Comment:
emit is needed

Reply:
fixed + added tests for AuctionCancelled event

Reply:
turns out it works without emit! but added anyway of course